nixos/modules/services/x11/xidlehook.nix

```diff
@@ -0,0 +1,225 @@
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      locker = mkOption {
+        default = "${pkgs.xlockmore}/bin/xlock"; # default according to `man xautolock`
+        example = "${pkgs.i3lock}/bin/i3lock -i /path/to/img";
+        type = types.str;
+
+        description = ''
+          The script to use when automatically locking the computer.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      notify = mkOption {
+        default = 10;
+        type = types.int;
+
+        description = ''
+          Time (in seconds) before the actual lock when the notification about the pending lock should be published.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      notifier = mkOption {
+        default = null;
+        example = "${pkgs.libnotify}/bin/notify-send \"Locking in ${cfg.notify} seconds\"";
+        type = types.nullOr types.str;
+
+        description = ''
+          Notification script to be used to warn about the pending autolock.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      killer = mkOption {
+        default = null; # default according to `man xautolock` is none
+        example = "${pkgs.systemd}/bin/systemctl suspend";
+        type = types.nullOr types.str;
+
+        description = ''
+          The script to use when nothing has happend for as long as <option>killtime</option>
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      killtime = mkOption {
+        default = 20; # default according to `man xautolock`
+        type = types.int;
+
+        description = ''
+          Minutes xidlehook waits until it executes the script specified in <option>killer</option>
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+    };
+  };
+
+  config = {
+    services.xserver.xidlehook.timers = mkDefault (
+      (optional cfg.enableNotifier {
+        duration = cfg.time * 60 - cfg.notify;
+        command = cfg.notifier;
+      }) ++ [{
+        duration = if cfg.enableNotifier then cfg.notify else cfg.time * 60;
+        command = cfg.locker;
+      }] ++ (optional (cfg.killer != null) {
+        duration = (cfg.killtime - cfg.time) * 60;
+        command = cfg.killer;
+      }));
+
+    environment.systemPackages = optional cfg.enable pkgs.xidlehook;
```

Comment:
Is this needed in the system environment? E.g. do users need to call the binary themselves for some reason? If not this can be removed

Reply:
This is necessary to get xidlehook-client in $PATH, which is the equivalent of xscreensaver-command & is used for things like lock-now keyboard shortcuts.